### DISCUSSION TITLE
How can I get "optgroup" value on Select element?

### DISCUSSION THREAD

**Discussion by devhe4d** [react-hook-form/react-hook-form#7045] [Q&A]
Hey everyone!

I wonder how can I get access to the optgroup value on the submit? I tried to {...register } it, but it didn't work.
say we are in a situation like this and we need to know under which option-group user selected their choice:

```
<select {...register('selected.item')}>
          <optgroup label="Fruit" {...register('selected.group')}>
            <option value="orange">Orange</option>
            <option value="apple">Apple</option>
          </optgroup>

          <optgroup label="Food" {...register('selected.group')}>
            <option value="cheese">Cheese</option>
            <option value="pasta">Pasta</option>
          </optgroup>
        </select>
```

any idea?

**Comment by JamesWong1999** [comment]
can you provide a codesandbox? it should just worked.

**Reply by devhe4d** [reply]
Sure, I have created a demo [here](https://stackblitz.com/edit/nextjs-lfw5rt?file=pages/index.js)

**Comment by devhe4d** [comment]
@bluebill1049 could you please take a look at this? Thanks :)

**Comment by bluebill1049** [comment]
isn't it working? https://codesandbox.io/s/amazing-leavitt-xy7q4?file=/src/App.js

**Reply by devhe4d** [reply]
No I'm still just getting the option value on submit without **optgroup** value

I want to get it like this: 

`selected: {
        item: 'orange',
        group: 'fruit'
}
`

**Reply by bluebill1049** [reply]
there is no such usage in native HTML with a select group, please refer to the documentation for native select.

**Comment by karnbibek** [comment]
Probably not an ideal solution but made this work by assigning the optgroup value as well to the the option by using a separator ("|" --> in my case)  and while submitting the form, split the value with the separator and get both values. E.g. shown below:


```
<select {...register('selected.item')}>
          <optgroup label="Fruit" {...register('selected.group')}>
            <option value={`Fruit|orange`}>Orange</option>
            <option value={`Fruit|apple`}>Apple</option>
          </optgroup>

          <optgroup label="Food" {...register('selected.group')}>
            <option value={`Food|cheese`}>Cheese</option>
            <option value={`Food|pasta`}>Pasta</option>
          </optgroup>
        </select>
        
        And while submitting the form:
         const onSubmit = async(data) => {
               const splitSelect = data.split("|");
console.log(split);
}
```
